File: C++/12_Integer_to_Roman.cpp
class Solution {
public:
    string intToRoman(int num) {
        int values[] =  {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        string romans[] =  {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        string res;
        for (int i = 0 ; i < 13 ; i++){
            while(num >= values[i]){
                num -= values[i];
                res += romans[i];
            }
        }
        return res;
    }
};

int stringToInteger(string input) {
    return stoi(input);
}

int main() {
    string line;
    while (getline(cin, line)) {
        int num = stringToInteger(line);
        
        string ret = Solution().intToRoman(num);

        string out = (ret);
        cout << out << endl;
    }
    return 0;
}

File: python/12_Integer_to_Roman.py
# I             1
# V             5
# X             10
# L             50
# C             100
# D             500
# M             1000

class Solution:
    def intToRoman(self, num: int) -> str:
        numMap = {'I':1,'IV':4,'V':5,'IX':9,'X':10,'XL':40,'L':50,'LC':90,'C':100,'CD':400,'D':500,'CM':900,'M':1000}
        numbers = [1000,900,500,400,100,90,50,40,10,9,5,4,1]
        romans = ['M','CM','D','CD','C','XC','L','XL','X','IX','V','IV','I']
        res = ''
        num_left = num
        def substitute(res,num_left):
            for i in range(len(numbers)):
                if num_left < numbers[i]:continue
                elif num_left == numbers[i]:
                    return res+romans[i]
                else:
                    res += romans[i]
                    num_left -= numbers[i]
                    return substitue(res,num_left)
        
        return substitute(res,num)